src/views/refund/refundActions.js

```diff
@@ -110,73 +103,54 @@ export const startRefund = (
       .then(response => {
         dispatch(
           getFeeEstimation(feeEstimation => {
-            const {
-              refundTransaction,
-              lockupTransactionId,
-            } = createRefundTransaction(
+            const transaction = refundTransaction(
               refundFile,
               response,
               destinationAddress,
               currency,
               feeEstimation
             );
 
-            dispatch(setRefundTransactionHash(refundTransaction.getId()));
+            dispatch(setRefundTransactionHash(transaction.getId()));
             dispatch(
-              broadcastRefund(
-                currency,
-                refundTransaction.toHex(),
-                lockupTransactionId,
-                () => {
-                  dispatch(refundResponse(true, response.data));
-
-                  cb();
-                }
-              )
+              broadcastRefund(currency, transaction.toHex(), () => {
+                dispatch(refundResponse(true, response.data));
+
+                cb();
+              })
             );
           })
         );
       })
       .catch(error => {
         const message = error.response.data.error;
 
-        window.alert(`Failed to refund swap: ${message}`);
+        notificationManager.spawnNotification(
+          'Failed to refund swap:',
+          message
+        );
         dispatch(refundResponse(false, message));
       });
   };
 };
 
-const broadcastRefund = (currency, transactionHex, lockupTransactionId, cb) => {
+const broadcastRefund = (currency, refundTransaction, cb) => {
   const url = `${boltzApi}/broadcasttransaction`;
   return dispatch => {
     dispatch(refundRequest());
     axios
       .post(url, {
         currency,
-        transactionHex,
+        transactionHex: refundTransaction,
       })
       .then(() => cb())
-      .catch(response => {
-        const error = response.response.data.error;
-        let message = `Failed to broadcast refund transaction: ${error}`;
-
-        if (
-          error ===
-          'please wait until your lockup transaction has 10 confirmations before you try to refund'
-        ) {
-          message +=
-            '. Click OK to open the lockup transaction in a block explorer';
```

Comment:
This feature is missing in the new implementation. Is it possible to redirect to a new tab with the new kind of notifications?